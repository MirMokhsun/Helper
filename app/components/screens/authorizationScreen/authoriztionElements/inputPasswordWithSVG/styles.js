import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        height: 50,
        flexDirection: 'column',
        alignItems: 'center',
        width: 290,
    },
    textinputWithTextAlightRight: {
        textAlign: 'right',
    },
    warningStyle: {
        borderColor: "#F86F6F"
    },
    borderGray: {
        borderColor: "#D3D9E7"
    },
    textinputwrapper: {
        flexDirection: 'row',
        backgroundColor: '#FFFFFF',
        width: '100%',
        height: 35,
        alignItems: 'center',
        justifyContent: 'space-between',
        borderBottomWidth: 1,
        borderColor: "#1459D2",
        paddingRight: 5,
    },
    textinput: {
        flex: 1,
        padding: 0,
        height: "100%",
        backgroundColor: 'white',
        fontFamily: 'Roboto-Light',
        fontSize: 18,
        color: "#1459D2",
        marginRight: 10,
    },
});