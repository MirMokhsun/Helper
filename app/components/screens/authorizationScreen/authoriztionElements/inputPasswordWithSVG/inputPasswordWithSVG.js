import React from 'react';
import { View, TouchableOpacity, TextInput } from 'react-native';
import { styles } from './styles';
import Dontshow from '../../../../commonComponents/svg/Dontshow';
import Show from '../../../../commonComponents/svg/Show';
import { WarningInputText } from '../../../../commonComponents/commonInput/commonInputElements/warningInputText';
import { validationTest } from '../../../../../api/regexFilters/validationInput';

export class InputPasswordWithSVG extends React.Component {
    constructor(props) {
        super(props);
        this._password = "";
        this.state = {
            isSecureTextEntry: true,
            isShowWarning: false,

        }
    }

    setSecureTextEntry = () => {
        const { isSecureTextEntry } = this.state;

        this.setState({
            isSecureTextEntry: !isSecureTextEntry,
        });
    }

    checkValue = (text) => {
        const { filter, action } = this.props;
        this._value = text;
        const isValidValue = validationTest(this._value, filter);
        action({ isValid: isValidValue, value: this._value.trim() });
        if (isValidValue) {
            this.setState({
                isShowWarning: false,
            });
        }
    }

    onBlur = () => {
        const { isValid } = this.props;
        if ( !isValid ) {
            this.setState({
                isShowWarning: true,
            });
        }
    }

    render() {

        const { currentLanguageName, placeholder, errorText, testID, textErrorID, keyboardType } = this.props;
        const { isShowWarning, isSecureTextEntry } = this.state;
        return (
            <View style={styles.container}>
                <View style={[styles.textinputwrapper, this._value ? null : styles.borderGray, isShowWarning ? styles.warningStyle : null,]}>
                    <TextInput
                        secureTextEntry={isSecureTextEntry}
                        keyboardType={keyboardType || 'default'}
                        testID={testID}
                        accessibilityLabel={testID}
                        value={this._value}
                        onBlur={this.onBlur}
                        selectionColor="#5A6772"
                        placeholderTextColor='#9FA0B2'
                        underlineColorAndroid='transparent'
                        onChangeText={this.checkValue}
                        style={[styles.textinput, currentLanguageName === 'ar' ? styles.textinputWithTextAlightRight : null]}
                        placeholder={placeholder} />
                    <TouchableOpacity onPress={this.setSecureTextEntry} testID="BtnShowPassAuthorizationID" accessibilityLabel="BtnShowPassAuthorizationID">
                        {isSecureTextEntry ? <Dontshow /> : <Show />}
                    </TouchableOpacity >
                </View>
                <WarningInputText isShowWarningText={isShowWarning} text={errorText} testID={textErrorID} />
            </View>
        )
    }
}

export default InputPasswordWithSVG;