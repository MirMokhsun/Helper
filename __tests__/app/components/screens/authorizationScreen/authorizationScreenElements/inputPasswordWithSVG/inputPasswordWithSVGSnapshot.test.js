import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { InputPasswordWithSVG } from '../../../../../../../app/components/screens/authorizationScreen/authoriztionElements/inputPasswordWithSVG/inputPasswordWithSVG';

it('create snapshot InputPasswordWithSVG', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<InputPasswordWithSVG
        placeholder={'password'}
        errorText={"Password failed"}
        filter={/$/}
        action={()=>{}}
        isValid={true}
        value={'value'}
        currentLanguageName={'ru'}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});