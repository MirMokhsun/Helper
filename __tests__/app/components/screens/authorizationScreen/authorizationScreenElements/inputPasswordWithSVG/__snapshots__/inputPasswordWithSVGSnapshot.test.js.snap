// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot InputPasswordWithSVG 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "height": 50,
      "width": 290,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "borderColor": "#1459D2",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
          "paddingRight": 5,
          "width": "100%",
        },
        Object {
          "borderColor": "#D3D9E7",
        },
        null,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="password"
      placeholderTextColor="#9FA0B2"
      rejectResponderTermination={true}
      secureTextEntry={true}
      selectionColor="#5A6772"
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "color": "#1459D2",
            "flex": 1,
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "height": "100%",
            "marginRight": 10,
            "padding": 0,
          },
          null,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TouchableOpacity
      accessibilityLabel="BtnShowPassAuthorizationID"
      activeOpacity={0.2}
      onPress={[Function]}
      testID="BtnShowPassAuthorizationID"
    >
      <Dontshow />
    </TouchableOpacity>
  </View>
  <WarningInputText
    isShowWarningText={false}
    text="Password failed"
  />
</View>
`;
